src/app/units/states/edit/directives/unit-group-set-editor/unit-group-set-editor.coffee

```diff
@@ -40,6 +41,31 @@ angular.module('doubtfire.units.states.edit.directives.unit-group-set-editor', [
         (response) -> alertService.add("danger", "Failed to update group set. #{response.data.error}", 6000)
       )
 
+    $scope.toggleLocked = (gs) ->
+      GroupSet.update(
+        {
+          unit_id: $scope.unit.id
+          id: gs.id
+          group_set:
+            {
+              locked: !gs.locked
+            }
+        }
+
+        (response) ->
+          alertService.add("success", "#{if response.locked then 'Locked' else 'Unlocked'} #{gs.name}", 2000)
+          gs.locked = response.locked
+          $scope.$broadcast 'UnitGroupSetEditor/SelectedGroupSetChanged', { id: gs.id }
+
+        (response) ->
+          alertService.add("danger", "Failed to #{if gs.locked then 'unlock' else 'lock'} #{gs.name}", 6000)
+      )
+
+    # Listen for groups all groups being locked.
+    listeners.push $scope.$on 'GroupSelector/GroupLockToggled', (e, args) ->
+      if args.groupSetId == $scope.selectedGroupSet.id
+        $scope.selectedGroupSet.locked = args.allLocked
```

Comment:
Should only be when true - as a group set may be locked when all groups are not locked.

Reply:
Context about the changes that are requested:

> What I would like is to also have a lock on the group set
> 
> So each group is locked if it is locked OR the group set is locked
> 
> To lock all groups you then just lock the group set
> 
> Simplifies the check for all groups being locked - as you check just the group set
> 
> Also means you can have some individual groups locked, and then lock the group set, if you unlock the group set then those that were originally locked will remain locked

Reply:
We have made the changes in a new commit, which can be viewed here: https://github.com/doubtfire-lms/doubtfire-web/pull/281/commits/2d683e9ffc305b143aad3a991d3146fc35ce27e9.

Thank you!

Reply:
The corresponding backend commit is: https://github.com/doubtfire-lms/doubtfire-api/pull/284/commits/7af266b4e18de04490c880e3790c712242ee0a4a

A video demonstrating the effect of this change is at https://youtu.be/faWO8D796Wo